flink-core/src/main/java/org/apache/flink/util/ClassLoaderWithErrorHandler.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+import java.util.function.Consumer;
+
+/**
+ * This {@link URLClassLoader} decorator accepts a custom exception handler.
+ *
+ * <p>The handler is called if an exception occurs in the {@link #loadClass(String, boolean)} of inner class loader.
+ */
+public class ClassLoaderWithErrorHandler extends URLClassLoader {
```

Comment:
We could name this class 

```suggestion
public class FlinkUserCodeClassLoader extends URLClassLoader {
```

if it is the base class for every Flink user code class loader.